ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -1591,7 +1588,7 @@ checkKnownInvalid ::
   -> InternalEnv m blk arrival judgment
   -> Header blk
   -> m ()
-checkKnownInvalid cfgEnv dynEnv intEnv hdr = case scrutinee of
+checkKnownInvalid cfgEnv dynEnv intEnv hdr = case headerPrevHash hdr of
```

Comment:
Again, I think we want a comment here that calls out that this drop 1 exists because of diffusion pipelining

Reply:
Do we want to restore [this](https://github.com/IntersectMBO/ouroboros-consensus/pull/1262/files#diff-2f5ac08e9601d9f879119ec18969c373486813f7448c70d053c5e4ff70d0f17fL1617) comment (with some modifications)?

```
    -- When pipelining, the tip of the candidate is forgiven for being an￼
    -- invalid block, but not if it extends any invalid blocks.
    -- Disconnect if the parent block of `hdr` is known to be invalid.		
```

Reply:
Resolved as in https://github.com/IntersectMBO/ouroboros-consensus/pull/1262#discussion_r1798970918